### PR TITLE
refactor: Update `CosmosTheme` to support brightness and dark/light theme as constructor parameters

### PR DESCRIPTION
**Pull request by wal33d006**

### CODE REVIEW

File: packages/cosmos_ui_components/lib/cosmos_theme.dart [hunk 1]
```diff
@@ -87,6 +88,68 @@ class CosmosThemeData extends Equatable {
...
+    this.borderRadiusS = defaultBorderRadiusS,
+    this.elevationS = defaultElevationS,
+    this.elevationM = defaultElevationM,
+    this.elevationL = defaultElevationL,
+    this.colors = darkThemeColors,
+    this.brightness = Brightness.dark,
+    this.errorColor = CosmosColorsData.defaultError,
+  });
+
+  const CosmosThemeData.light({
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/cosmos_theme.dart:122]: a lot of redundancy here, you can provide one default constructor with all those default values and reuse that, like so:
```dart
  
const CosmosThemeData({
    required this.colors,
    required this.brightness,
    this.spacingXXXL = defaultSpacingXXXL,
    this.spacingXXL = defaultSpacingXXL,
    this.spacingXL = defaultSpacingXL,
    this.spacingL = defaultSpacingL,
    this.spacingM = defaultSpacingM,
    this.spacingS = defaultSpacingS,
    this.spacingXS = defaultSpacingXS,
    this.longDuration = defaultLongDuration,
    this.mediumDuration = defaultMediumDuration,
    this.shortDuration = defaultShortDuration,
    this.radiusXL = defaultRadiusXL,
    this.radiusL = defaultRadiusL,
    this.radiusM = defaultRadiusM,
    this.radiusS = defaultRadiusS,
    this.fontSizeS = defaultFontSizeS,
    this.fontSizeM = defaultFontSizeM,
    this.fontSizeL = defaultFontSizeL,
    this.fontSizeXL = defaultFontSizeXL,
    this.fontSizeXXL = defaultFontSizeXXL,
    this.borderRadiusL = defaultBorderRadiusL,
    this.borderRadiusM = defaultBorderRadiusM,
    this.borderRadiusS = defaultBorderRadiusS,
    this.elevationS = defaultElevationS,
    this.elevationM = defaultElevationM,
    this.elevationL = defaultElevationL,
    this.errorColor = CosmosColorsData.defaultError,
  });

  const CosmosThemeData.dark() : this(colors: darkThemeColors, brightness: Brightness.dark);
  
  const CosmosThemeData.light() : this(colors: lightThemeColors, brightness: Brightness.light);
 ```

File: packages/cosmos_ui_components/lib/cosmos_theme.dart [hunk 2]
```diff
@@ -87,6 +88,68 @@ class CosmosThemeData extends Equatable {
...
+    this.fontSizeXXL = defaultFontSizeXXL,
+    this.borderRadiusL = defaultBorderRadiusL,
+    this.borderRadiusM = defaultBorderRadiusM,
+    this.borderRadiusS = defaultBorderRadiusS,
+    this.elevationS = defaultElevationS,
+    this.elevationM = defaultElevationM,
+    this.elevationL = defaultElevationL,
+    this.colors = const CosmosColorsData(),
+    this.brightness = Brightness.light,
+    this.errorColor = CosmosColorsData.defaultError,
```
Comment (andrzejchm) [inline packages/cosmos_ui_components/lib/cosmos_theme.dart:150]: btw. this errorColor should be part of `CosmosColorData`

**Review by andrzejchm — CHANGES_REQUESTED** [review]

File: packages/cosmos_ui_components/lib/cosmos_theme.dart (hunk 2 above)
Comment (wal33d006) [reply packages/cosmos_ui_components/lib/cosmos_theme.dart:150]: This was a part of `CosmosColorsData` but we it had the same value for both themes, so we decided it to keep it a part of `CosmosTheme`

Comment (andrzejchm) [reply packages/cosmos_ui_components/lib/cosmos_theme.dart:150]: hm, not sure I get it, even if its the same for both themes, it still can reside in `CosmosColorData`, as this is the place for colors basically

File: packages/cosmos_ui_components/lib/cosmos_theme.dart [hunk 3]
```diff
@@ -84,9 +87,14 @@ class CosmosThemeData extends Equatable {
...
     this.colors = const CosmosColorsData(),
     this.brightness = Brightness.light,
-    this.errorColor = CosmosColorsData.defaultError,
   });
 
+  const CosmosThemeData.dark() : this(colors: darkThemeColors, brightness: Brightness.dark);
+
+  const CosmosThemeData.light() : this(colors: lightThemeColors, brightness: Brightness.light);
+
+  ThemeData buildFlutterTheme() => convertCosmosThemeToMaterialTheme(this);
```
Comment (wal33d006) [inline packages/cosmos_ui_components/lib/cosmos_theme.dart:96]: Added this function as per our discussion. Do let me know if this is okay. I couldn't figure out a way to pass the CosmosTheme to this converter.

Comment (wal33d006) [reply packages/cosmos_ui_components/lib/cosmos_theme.dart:96]: ^@andrzejchm

**Review by andrzejchm — CHANGES_REQUESTED** [review]
just this one last comment and we're good to go:
https://github.com/tendermint/flutter/pull/290#discussion_r875588628

**Pull request merged**